src/System.Drawing.Common/src/System/Drawing/Icon.Unix.cs

```diff
@@ -704,6 +704,11 @@ public int Height
         {
             get
             {
+                if (disposed)
```

Comment:
does Windows have a `CheckIfDisposed` helper or similar? We could consolidate this code

Reply:
I could add one.

Reply:
Actually, the two implementations seem to have slightly different notions of disposed-ness. The Windows version always stores and tracks a native handle, which is only null if the object has been disposed. The Unix version tracks whether the Icon has been disposed independently, and only creates an actual handle when it is requested.